django_outbox_pattern/consumers.py

```diff
@@ -34,30 +34,39 @@ def __init__(self, connection, username, passcode):
         self.set_listener(self.listener_name, self.listener_class(self))
 
     def message_handler(self, body, headers):
+        db.close_old_connections()
         try:
             body = json.loads(body)
         except json.JSONDecodeError as exc:
             logger.exception(exc)
 
         payload = Payload(self.connection, body, headers)
-        received, created = self.received_class.objects.get_or_create(msg_id=_get_msg_id(headers))
-        if created:
-            received.body = body
-            received.headers = headers
-            try:
-                self.callback(payload)
-            except Exception as exc:  # pylint: disable=broad-except
-                received.status = StatusChoice.FAILED
-                if self.is_connected():
-                    payload.nack()
-                logger.exception(exc)
-            else:
-                received.status = StatusChoice.SUCCEEDED
-                payload.ack()
-            finally:
-                received.save()
-        else:
+        message_id = _get_msg_id(headers)
+
+        if self.received_class.objects.filter(msg_id=message_id).exists():
+            db.close_old_connections()
```

Comment:
[question]
Does it make sense to close again the connection the connection here, since you've closed on line 37?

Reply:
It's because it opens a new connection on line 46 :)

Reply:
But the one on line 37 is useless, I just removed it. Good observation!